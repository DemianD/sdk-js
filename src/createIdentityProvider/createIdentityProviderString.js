function encodeParams(options = {}) {
  return Object.keys(options)
    .map(key => `${key}=${encodeURIComponent(options[key])}`)
    .join('&')
}

export default function createIdentityProviderString(type, options) {
  return `${type}://${encodeParams(options)}`
}
