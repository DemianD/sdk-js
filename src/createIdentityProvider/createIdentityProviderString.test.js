import createIdentityProviderString from './createIdentityProviderString'

describe('createIdentityProviderString', () => {
  it('should be able to create an identity provider string', () => {
    expect(
      createIdentityProviderString('client', {
        Foo: 'Foo',
        Bar: 'Bar',
        Baz: 'http://foo.bar.baz/',
      })
    ).toMatchSnapshot()
  })

  it('should be possible to pass no options at all', () => {
    expect(createIdentityProviderString('client')).toMatchSnapshot()
  })

  it('should filter out undefined values', () => {
    expect(
      createIdentityProviderString('client', {
        Foo: 'Foo',
        Bar: 'Bar',
        Baz: undefined,
      })
    ).toMatchSnapshot()
  })
})
