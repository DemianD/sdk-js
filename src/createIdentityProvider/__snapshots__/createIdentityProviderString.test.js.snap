// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createIdentityProviderString should be able to create an identity provider string 1`] = `"client://Foo=Foo&Bar=Bar&Baz=http%3A%2F%2Ffoo.bar.baz%2F"`;

exports[`createIdentityProviderString should be possible to pass no options at all 1`] = `"client://"`;

exports[`createIdentityProviderString should filter out undefined values 1`] = `"client://Foo=Foo&Bar=Bar"`;
